### DISCUSSION TITLE
Select all, Contents, Zoom in/out and Confirm to save

### DISCUSSION THREAD

**Discussion by guanglinn** [gsantner/markor#2095] [Ideas] [answered]
Here are some new features in my develop branch. Do you think it is necessary to request to add?

1. Select all files

![markor_select_all.gif](https://github.com/gsantner/markor/assets/60415467/f02af3c4-00d8-48e5-b9e3-bafd1875cd4d)

2. Contents in view mode

![markor_contents.gif](https://github.com/gsantner/markor/assets/60415467/b5cc6136-318d-4425-826d-7bb2fd492b07)

3. Zoom in/out in view mode

![markor_zoom.gif](https://github.com/gsantner/markor/assets/60415467/de23eda1-e2fe-4863-b0fe-9ce4afb36cf5)

4. Confirm to save changes

![Screenshot_20230817_235326_net.gsantner.markor_test.jpg](https://github.com/gsantner/markor/assets/60415467/7a714992-bbe6-468e-a79e-c8695ebcbca0)

**Comment by gsantner** [comment, marked as answer]
Hello @guang-lin,

you mean if you should open separate issues for them? I don't think so, OK to talk at once here, seemingly smaller things.

1) Fine I guess, but make sure it's in the overflow (...) menu 

2) Huh, I thought we already have a TOC navigation for view mode. Did you run across something existing?

3) The feature itself is helpful I think, but please don't add new always-visible buttons to the top bar. Space is very limited there. Overflow menu or bottom TextAction buttons would be an option.

4) I'm not keen on that one. Saving and file management is very very troublesome and not wanna require some manual action.

**Reply by guanglinn** [reply]
Thanks for your suggestions.
Perhaps these could provide some helpful references for others.

**My supplements**

2, Actually, I don't really like to use TOC, I think it takes up space and needs to scroll to the top, even at the bottom of the document. I don't know if some people also have this need;

4, In my codes, this is based on existing logic for saving, if Android kills Markor in the background,  the data will still be saved timely. Besides, it doesn't ask if saving changes in the QuickNote and To-Do.
